qwerty_ast/src/ast.rs

```diff
@@ -349,6 +349,74 @@ impl<E> FunctionDef<E> {
     }
 }
 
+// NOTE: Until there is a nice way to distinguish these, there
+// are two display implementations. One could use Trivializable,
+// but you would have to modify it to return Option<Expr> and
+// this seemed antithetical to the purpose. I also didn't want
+// to create another trait just for this.
+impl std::fmt::Display for FunctionDef<qpu::Expr> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "fn {}(", self.name)?;
```

Comment:
Can we use the syntax from the Python DSL? The existing `fmt::Display` implementations always do
```suggestion
        write!(f, "def {}(", self.name)?;
```